CI test summary fails when artifact upload does not produce artifact-url

## Summary

Some CI test jobs can fail after all tests pass because `run-tests.yml` always passes `-u "${{ steps.upload-logs.outputs.artifact-url }}"` to `GenerateTestSummary`. When the `Upload logs, and test results` step fails before producing `artifact-url`, the summary command receives an empty URL value and exits with:

```text
Required argument missing for option: '-u'.
```

This makes otherwise-passing test jobs red and contributes to recurring CI noise.

## Example

Run: https://github.com/microsoft/aspire/actions/runs/24969013933/job/73108901606

The test step passed:

```text
Test run summary: Passed!
  total: 57
  failed: 0
  succeeded: 57
```

The post-test steps then failed:

```text
Upload logs, and test results
With the provided path, there will be 4 files uploaded
Artifact name is valid!
Root directory input is valid!
```

```text
Generate test results summary
Required argument missing for option: '-u'.

Usage:
  GenerateTestSummary <dirPathOrTrxFilePath> [options]

Options:
  -u, --url <url>        URL for test links

Error: Process completed with exit code 1.
```

The screenshot provided by the reporter shows the same failure chain: `Upload logs, and test results` fails, `Generate test results summary` runs, and `GenerateTestSummary` fails because `-u` has no value.

## Occurrences found

I found 6 confirmed occurrences of this form in the last 7 days while scanning CI failures:

| Run | Job |
| --- | --- |
| https://github.com/microsoft/aspire/actions/runs/24969013933 | https://github.com/microsoft/aspire/actions/runs/24969013933/job/73108901606 |
| https://github.com/microsoft/aspire/actions/runs/24955382944 | https://github.com/microsoft/aspire/actions/runs/24955382944/job/73072795987 |
| https://github.com/microsoft/aspire/actions/runs/24950687678 | https://github.com/microsoft/aspire/actions/runs/24950687678/job/73060280005 |
| https://github.com/microsoft/aspire/actions/runs/24947077611 | https://github.com/microsoft/aspire/actions/runs/24947077611/job/73050524718 |
| https://github.com/microsoft/aspire/actions/runs/24924310648 | https://github.com/microsoft/aspire/actions/runs/24924310648/job/72991697378 |
| https://github.com/microsoft/aspire/actions/runs/24918308249 | https://github.com/microsoft/aspire/actions/runs/24918308249/job/72974985166 |

The count is conservative: the scan confirmed these by matching failed post-test steps and/or the `Required argument missing for option: '-u'.` log signature.

## Suspected cause

`.github/workflows/run-tests.yml` currently has:

```powershell
& ${{ env.DOTNET_SCRIPT }} run --project "${{ github.workspace }}/tools/GenerateTestSummary/GenerateTestSummary.csproj" -- "${{ github.workspace }}/testresults" -u "${{ steps.upload-logs.outputs.artifact-url }}"
```

When the upload step fails before setting `artifact-url`, the expression expands to an empty string. `System.CommandLine` then treats `-u` as present without its required value and fails the job.

## Possible fix

Only pass `-u` when `steps.upload-logs.outputs.artifact-url` is non-empty, or make `GenerateTestSummary` tolerate an omitted/empty URL and generate a summary without links.

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -642,5 +642,18 @@ jobs:
         shell: pwsh
         run: |
           if (Test-Path "${{ github.workspace }}/testresults") {
-            & ${{ env.DOTNET_SCRIPT }} run --project "${{ github.workspace }}/tools/GenerateTestSummary/GenerateTestSummary.csproj" -- "${{ github.workspace }}/testresults" -u "${{ steps.upload-logs.outputs.artifact-url }}"
+            $summaryArgs = @(
+              "run",
+              "--project",
+              "${{ github.workspace }}/tools/GenerateTestSummary/GenerateTestSummary.csproj",
+              "--",
+              "${{ github.workspace }}/testresults"
+            )
+
+            $artifactUrl = "${{ steps.upload-logs.outputs.artifact-url }}"
+            if (-not [string]::IsNullOrWhiteSpace($artifactUrl)) {
+              $summaryArgs += @("-u", $artifactUrl)
+            }
+
+            & ${{ env.DOTNET_SCRIPT }} @summaryArgs
           }

